qiskit_experiments/library/characterization/cr_hamiltonian_analysis.py

```diff
@@ -0,0 +1,298 @@
+        )
+        default_options.ylim = (-1, 1)
+
+        return default_options
+
+    def _setup_fitting(self, **extra_options) -> Union[Dict[str, Any], List[Dict[str, Any]]]:
+        """Fitter options."""
+        user_p0 = self._get_option("p0")
+        user_bounds = self._get_option("bounds")
+
+        bounds = {
+            "px0": user_bounds["px0"] or (-np.inf, np.inf),
+            "py0": user_bounds["py0"] or (-np.inf, np.inf),
+            "pz0": user_bounds["pz0"] or (-np.inf, np.inf),
+            "px1": user_bounds["px1"] or (-np.inf, np.inf),
+            "py1": user_bounds["py1"] or (-np.inf, np.inf),
+            "pz1": user_bounds["pz1"] or (-np.inf, np.inf),
+            "b": user_bounds["b"] or (-1, 1),
+        }
+
+        guesses = defaultdict(list)
+
+        for control in (0, 1):
+            # start from Z oscillation
+            x_data = self._data(series_name=f"x|c={control}")
+            y_data = self._data(series_name=f"y|c={control}")
+            z_data = self._data(series_name=f"z|c={control}")
+
+            omega_xyz = []
+            for data in (x_data, y_data, z_data):
+                fft_freq = curve.guess.frequency(data.x, data.y)
+                # oscillation amplitude might be almost zero, then exclude from average
+                if fft_freq > 0:
+                    omega_xyz.append(fft_freq)
+            if omega_xyz:
+                omega = 2 * np.pi * np.average(omega_xyz)
+            else:
+                omega = 0.0
+
+            zmin, zmax = np.percentile(z_data.y, [10, 90])
+            zrange = zmax - zmin
+            if zrange > 2.0:
+                theta = 0.0
+            else:
+                theta = np.arccos(np.sqrt(zrange / 2))
+
+            # The FFT might be up to 1/2 bin off
+            df = 1 / ((z_data.x[1] - z_data.x[0]) * len(z_data.x))
+            for omega_shifted in [omega, omega - df / 2, omega + df / 2]:
+                for phi in np.linspace(-np.pi, np.pi, 9):
+                    px = omega_shifted * np.cos(theta) * np.cos(phi)
+                    py = omega_shifted * np.cos(theta) * np.sin(phi)
+                    pz = omega_shifted * np.sin(theta)
+                    guesses[control].append(
+                        {
+                            f"px{control}": user_p0[f"px{control}"] or px,
+                            f"py{control}": user_p0[f"py{control}"] or py,
+                            f"pz{control}": user_p0[f"pz{control}"] or pz,
+                        }
+                    )
+
+        fit_options = []
+        for p0s, p1s in zip(guesses[0], guesses[1]):
+            fit_option = {
+                "p0": {**p0s, **p1s, "b": user_p0["b"] or 1e-9},
+                "bounds": bounds,
+            }
+            fit_option.update(extra_options)
+            fit_options.append(fit_option)
+
+        return fit_options
+
+    def _evaluate_quality(self, fit_data: curve.FitData) -> Union[str, None]:
```

Comment:
should we have a criteria on chi-squared?

Reply:
I realized that multi-object function fit tends to report large chi-squared value (as you can see in the attached figure). I think what we want to know with the experiment is Pauli coefficient rather than fit params, so important is uncertainty of coefficients.

Reply:
Sounds good.